server/collab/vectors.py

```diff
@@ -0,0 +1,60 @@
+from collections import defaultdict
+
+
+class Vector:
+  @classmethod
+  def match(cls, soure, target):
+    raise NotImplementedError("Method match for vector type {} not "
+                              "implemented".format(cls))
+
+
+class DummyVector(Vector):
+  id = 'dummy'
+  name = 'Dummy'
+
+  @classmethod
+  def match(cls, soure, target):
+    return []
+
+
+class HashVector(Vector):
+  id = 'hash'
+  name = 'Hash'
```

Comment:
The idea is to remove `id` and `name` (not `HashVector` and `DummyVector`), specifically because that's an abstract class.